src/server/game/AI/SmartScripts/SmartScriptMgr.h

```diff
@@ -1479,15 +1473,13 @@ class TC_GAME_API SmartWaypointMgr
 
         void LoadFromDB();
 
-        WPPath* GetPath(uint32 id)
+        WPPath const &GetPath(uint32 id)
         {
-            if (waypoint_map.find(id) != waypoint_map.end())
-                return waypoint_map[id];
-            else return nullptr;
+            return waypoint_map[id];
```

Comment:
This is a bad change, what if someone requests a path not present in waypoint_map?